src/stores/RootStore.js

```diff
@@ -0,0 +1,23 @@
+// @flow
```

Comment:
Turns out RootStore pattern has its own pitfalls. If I want to use one store reference into another store, I have to initalize the first store before the second...For example, I'm using ErrorStore inside `autorun` in AuthStore, if I initalize AuthStore before ErorrStore, it throws.

Reply:
Singletons are fine. You can get the dependency injection working by splitting up stores but I don’t think it’s worth it in JavaScript